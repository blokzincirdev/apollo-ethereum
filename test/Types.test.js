const gql = require('graphql-tag')

const Types = artifacts.require('Types')

const config = {
  contracts: {
    Types: true,
  },
}

const QUERY_TYPES = gql`
  query($address: Address) {
    Types(address: $address) {
      bigNumberA
      bigNumberB
      integer
      bytesA
      bytesB
      stringType
      firstBool: bools(index: "0")
      lastInt: ints(index: "4")
      allBools
      allInts
      tlaScore(key: "0x637461")
      createdAt
    }
  }
`

contract('Types', () => {
  let client

  before(async () => {
    client = await createClient(config, { Types })
  })

  it('generates graphql schema', async () => {
    expect(client.source).to.matchSnapshot()
  })

  it('returns the correct types', async () => {
    const variables = { address: Types.address }
    const data = await client.execute(QUERY_TYPES, variables)
    expect(data).to.matchSnapshot()
  })
})
