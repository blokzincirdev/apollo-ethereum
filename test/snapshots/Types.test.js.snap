// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contract: Types generates graphql schema 1`] = `
"directive @erc1820(interfaceName: String) on FIELD_DEFINITION

directive @mappingIndex(mapping: String) on FIELD_DEFINITION

directive @contract(field: String) on FIELD_DEFINITION

scalar Address

scalar BigNumber

scalar Bytes

type Query {
  Types(address: Address!): Types
}

type Types {
  _address: Address
  bigNumberA: BigNumber
  bigNumberB: BigNumber
  boolean: Boolean
  bools(index: BigNumber!): Boolean
  bytesA: Bytes
  bytesB: Bytes
  integer: Int
  ints(index: BigNumber!): Int
  stringType: String
  tlaScore(key: Bytes!): BigNumber
  allBools: [Boolean]
  allInts: [Int]
}
"
`;

exports[`Contract: Types returns the correct types 1`] = `
Object {
  "Types": Object {
    "allBools": Array [
      true,
      false,
      true,
    ],
    "allInts": Array [
      1,
      2,
      3,
      4,
      5,
    ],
    "bigNumberA": BigNumber {
      "_hex": "0x2a",
    },
    "bigNumberB": BigNumber {
      "_hex": "0x38",
    },
    "bytesA": "0x666f6f",
    "bytesB": "0x64796e616d6963616c6c792073697a656420627974652073657175656e6365",
    "firstBool": true,
    "integer": 48,
    "lastInt": 5,
    "stringType": "dynamically sized string",
    "tlaScore": BigNumber {
      "_hex": "0x0512",
    },
  },
}
`;
